### ISSUE TITLE
Crash with Index -1 out of bounds for length 0 on QuiltFlower 1.7.0 executed via IntelliJ IDEA

### ISSUE THREAD

**Issue by KJP12** [QuiltMC/quiltflower#81]
Before we begin: I'm sorry, but I challenged myself to keep a minimal FizzBuzz entirely in the stack.

## Steps to Reproduce
1. Be a maniac willing to write raw Java bytecode using only the stack.
2. Create the following monstrosity: <details><summary>FizzBuzz.asm</summary>
   ```asm
   .class public FizzBuzz
   .super java/lang/Object
   .method public static main([Ljava/lang/String;)V
	.limit stack 4
	.limit locals 1
	getstatic java/lang/System/out Ljava/io/PrintStream;
	iconst_0
   loop:
	dup
	bipush 100
	isub
	ifgt exit
	iconst_1
	iadd
	dup
	iconst_3
	irem
	ifne skipFizz
	; Flip flops between the print stream
	swap
	dup
	ldc "Fizz"
	invokevirtual java/io/PrintStream/print(Ljava/lang/String;)V
	swap
	ineg
   skipFizz:
	dup
	ifgt alreadyPositive
	ineg
   alreadyPositive:
	dup
	iconst_5
	irem
	ifne skipBuzz
	swap
	dup
	ldc "Buzz"
	invokevirtual java/io/PrintStream/print(Ljava/lang/String;)V
	swap
	ineg
   skipBuzz:
	dup
	ifgt print
	ineg
	swap
	dup
	invokevirtual java/io/PrintStream/println()V
	swap
	goto loop
   print:
	; PrintStream, int -> int, PrintStream, int
	dup_x1
	invokestatic java/lang/Integer/toString(I)Ljava/lang/String;
	invokevirtual java/io/PrintStream/println(Ljava/lang/String;)V
	getstatic java/lang/System/out Ljava/io/PrintStream;
	swap
	goto loop
   exit:
	pop
	pop
	return
   .end method
   ```
   </details>
3. Use Jasmin to assemble it into the following class file: [FizzBuzz.zip](https://github.com/QuiltMC/quiltflower/files/8147694/FizzBuzz.zip) (This is a valid JAR; the entry point is `FizzBuzz`)
4. Toss the class file into IntelliJ for it to decompile.

## Expected Results
To get a probably mangled view of the "source" code.

## Actual results
The following crash:

<details><summary>The crash wrapped in the source code</summary>

```java
//
// Source code recreated from a .class file by Quiltflower
//

public class FizzBuzz {
    public static void main(String[] param0) {
        // $FF: Couldn't be decompiled
        // Please report this to the Quiltflower issue tracker, at https://github.com/QuiltMC/quiltflower/issues with a copy of the class file (if you have the rights to distribute it!)
        // java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
        //   at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
        //   at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
        //   at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
        //   at java.base/java.util.Objects.checkIndex(Objects.java:372)
        //   at java.base/java.util.ArrayList.get(ArrayList.java:459)
        //   at org.jetbrains.java.decompiler.modules.decompiler.ExprProcessor.processBlock(ExprProcessor.java:643)
        //   at org.jetbrains.java.decompiler.modules.decompiler.ExprProcessor.processStatement(ExprProcessor.java:156)
        //   at org.jetbrains.java.decompiler.main.rels.MethodProcessorRunnable.codeToJava(MethodProcessorRunnable.java:169)
        //
        // Bytecode:
        // 00: getstatic java/lang/System.out Ljava/io/PrintStream;
        // 03: bipush 0
        // 04: dup
        // 05: bipush 100
        // 07: isub
        // 08: ifgt 4c
        // 0b: bipush 1
        // 0c: iadd
        // 0d: dup
        // 0e: bipush 3
        // 0f: irem
        // 10: ifne 1c
        // 13: swap
        // 14: dup
        // 15: ldc "Fizz"
        // 17: invokevirtual java/io/PrintStream.print (Ljava/lang/String;)V
        // 1a: swap
        // 1b: ineg
        // 1c: dup
        // 1d: ifgt 21
        // 20: ineg
        // 21: dup
        // 22: bipush 5
        // 23: irem
        // 24: ifne 30
        // 27: swap
        // 28: dup
        // 29: ldc "Buzz"
        // 2b: invokevirtual java/io/PrintStream.print (Ljava/lang/String;)V
        // 2e: swap
        // 2f: ineg
        // 30: dup
        // 31: ifgt 3e
        // 34: ineg
        // 35: swap
        // 36: dup
        // 37: invokevirtual java/io/PrintStream.println ()V
        // 3a: swap
        // 3b: goto 04
        // 3e: dup_x1
        // 3f: invokestatic java/lang/Integer.toString (I)Ljava/lang/String;
        // 42: invokevirtual java/io/PrintStream.println (Ljava/lang/String;)V
        // 45: getstatic java/lang/System.out Ljava/io/PrintStream;
        // 48: swap
        // 49: goto 04
        // 4c: pop
        // 4d: pop
        // 4e: return
    }
}
```
</details>

**Comment by SuperCoder7979** [comment]
Ooh, this is a fun one. Thanks for the report, will look into!

**Comment by SuperCoder7979** [comment]
This should be fixed by df3bc52, released with 1.8.0. If it's not fixed, please re-open the issue!
